server/src/test/java/org/candlepin/model/PoolCuratorTest.java

```diff
@@ -528,6 +630,74 @@ public void testListBySourceEntitlement() {
 
     }
 
+
+    @Test
+    public void testListBySourceEntitlements() {
+        Pool sourcePool = TestUtil.createPool(owner, product);
+        Pool sourcePool2 = TestUtil.createPool(owner, product);
+        Pool sourcePool3 = TestUtil.createPool(owner, product);
+        poolCurator.create(sourcePool);
+        poolCurator.create(sourcePool2);
+        poolCurator.create(sourcePool3);
+        Entitlement e = new Entitlement(sourcePool, consumer, 1);
+        Entitlement e2 = new Entitlement(sourcePool2, consumer, 1);
+        Entitlement e3 = new Entitlement(sourcePool3, consumer, 1);
+        entitlementCurator.create(e);
+        entitlementCurator.create(e2);
+        entitlementCurator.create(e3);
+
+        Pool pool2 = TestUtil.createPool(owner, product);
+        pool2.setSourceEntitlement(e);
+        Pool pool3 = TestUtil.createPool(owner, product);
+        pool3.setSourceEntitlement(e);
+        Pool pool4 = TestUtil.createPool(owner, product);
+        pool4.setSourceEntitlement(e2);
+        Pool pool5 = TestUtil.createPool(owner, product);
+        pool5.setSourceEntitlement(e3);
+
+        poolCurator.create(pool2);
+        poolCurator.create(pool3);
+        poolCurator.create(pool4);
+        poolCurator.create(pool5);
+
+
+        List<Pool> pools = poolCurator.listBySourceEntitlements(Arrays.asList(e, e2));
+        assertEquals(3, pools.size());
+
+    }
+
+    @Test
+    public void retrieveFreeEntitlementsOfPools() {
+
+        Pool pool1 = TestUtil.createPool(owner, product);
+        poolCurator.create(pool1);
+        Entitlement ent11 = new Entitlement(pool1, consumer, 1);
+        entitlementCurator.create(ent11);
+        Entitlement ent12 = new Entitlement(pool1, consumer, 1);
+        entitlementCurator.create(ent12);
+        Entitlement ent13 = new Entitlement(pool1, consumer, 1);
+        entitlementCurator.create(ent13);
+
+        Pool pool2 = TestUtil.createPool(owner, product);
+        poolCurator.create(pool2);
+        Entitlement ent21 = new Entitlement(pool2, consumer, 1);
+        entitlementCurator.create(ent21);
+
+        Pool pool3 = TestUtil.createPool(owner, product);
+        poolCurator.create(pool3);
+        Entitlement ent31 = new Entitlement(pool3, consumer, 1);
+        entitlementCurator.create(ent31);
+
+        List<Entitlement> ents = poolCurator.retrieveFreeEntitlementsOfPools(Arrays.asList(pool1, pool2),
+                true);
+        assertEquals(4, ents.size());
+        assertTrue(ents.contains(ent11));
```

Comment:
You can get more informative error messages for failing tests by using

```
assertThat(ents, IsCollectionContaining.hasItem(ent1));
```

`assertTrue` is an assertion of last resort in my opinion since it just throws generic `java.lang.AssertionError` with zero context about what actually went wrong or was supposed to happen.